freeipa/src/main/java/com/sequenceiq/freeipa/flow/freeipa/cleanup/event/AbstractCleanupEvent.java

```diff
@@ -4,23 +4,34 @@
 
 public abstract class AbstractCleanupEvent extends CleanupEvent {
 
+    /**
+     * Need this for Jackson serialization.
+     */
+    private final CleanupEvent cleanupEvent;
+
     protected AbstractCleanupEvent(Long stackId) {
         super(stackId);
+        this.cleanupEvent = null;
     }
 
     public AbstractCleanupEvent(CleanupEvent cleanupEvent) {
         super(cleanupEvent.getResourceId(), cleanupEvent.getUsers(), cleanupEvent.getHosts(), cleanupEvent.getRoles(), cleanupEvent.getIps(),
                 cleanupEvent.getStatesToSkip(), cleanupEvent.getAccountId(), cleanupEvent.getOperationId(), cleanupEvent.getClusterName(),
                 cleanupEvent.getEnvironmentCrn());
+        this.cleanupEvent = cleanupEvent;
     }
 
     public AbstractCleanupEvent(String selector, CleanupEvent cleanupEvent) {
         super(selector, cleanupEvent.getResourceId(), cleanupEvent.getUsers(), cleanupEvent.getHosts(), cleanupEvent.getRoles(), cleanupEvent.getIps(),
                 cleanupEvent.getStatesToSkip(), cleanupEvent.getAccountId(), cleanupEvent.getOperationId(), cleanupEvent.getClusterName(),
                 cleanupEvent.getEnvironmentCrn());
+        this.cleanupEvent = cleanupEvent;
     }
 
-    private CleanupEvent getCleanupEvent() {
-        return this;
+    /**
+     * Need this for Jackson serialization. This must be public.
+     */
+    public CleanupEvent getCleanupEvent() {
+        return cleanupEvent;
```

Comment:
Actually, the original `private` method was not used anywhere, so I wonder if it could be simply dropped entirely.

Reply:
I myself added the original private method too. :)
Its presence is a must for Jackson serialization, as the comment says.

Reply:
Oh, I see. Then let us keep it this way. :smile: